mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala

```diff
@@ -291,6 +294,9 @@ class NaiveBayes private (
       if (!values.forall(_ >= 0.0)) {
         throw new SparkException(s"Naive Bayes requires nonnegative feature values but found $v.")
       }
+      if (modelType == "Bernoulli" && (!values.forall(v => v == 0.0 || v == 1.0) )) {
```

Comment:
This should be in a function separate from ```requireNonnegativeValues``` since it should be used in ```createCombiner``` but not ```mergeValue``` in the aggregation below.  I bet the test case with valid input would catch this issue if it had more rows with non-zeros.